agent/agent_test.go

```diff
@@ -1305,90 +1335,89 @@ task_groups:
 	}, []string{panicLog})
 }
 
-func (s *AgentSuite) TestTeardownTaskSucceeds() {
-	s.tc.taskGroup = "task_group_name"
+func (s *AgentSuite) TestTeardownTaskFails() {
+	const taskGroup = "task_group_name"
 	projYml := `
 task_groups:
   - name: task_group_name
+    teardown_task_can_fail_task: true
     teardown_task:
       - command: shell.exec
         params:
-          script: exit 0
+          script: exit 1
 `
 	s.setupRunTask(projYml)
-	s.tc.taskConfig.Task.TaskGroup = s.tc.taskGroup
+	s.tc.taskConfig.Task.TaskGroup = taskGroup
+	s.tc.taskConfig.TaskGroup = *s.tc.taskConfig.Project.FindTaskGroup(taskGroup)
 
-	s.NoError(s.a.runPostTaskCommands(s.ctx, s.tc))
+	s.Error(s.a.runPostTaskCommands(s.ctx, s.tc))
 
 	s.NoError(s.tc.logger.Close())
 	checkMockLogs(s.T(), s.mockCommunicator, s.tc.taskConfig.Task.Id, []string{
 		"Running post-task commands",
-		"Running command 'shell.exec' (step 1 of 1) in block 'teardown_task'",
-		"Finished command 'shell.exec' (step 1 of 1) in block 'teardown_task'",
-		"Finished running post-task commands",
-	}, []string{
-		panicLog,
-		"Set idle timeout for 'shell.exec'",
-	})
+	}, []string{panicLog})
 }
 
-func (s *AgentSuite) TestTeardownTaskFails() {
+func (s *AgentSuite) TestTeardownTaskTimeoutDoesNotFailTask() {
 	const taskGroup = "task_group_name"
-	s.tc.taskGroup = taskGroup
 	projYml := `
 task_groups:
   - name: task_group_name
-    teardown_task_can_fail_task: true
-    teardown_task:
+    teardown_task_timeout_secs: 1
+    setup_task:
       - command: shell.exec
         params:
-          script: exit 1
+          script: echo hi
 `
-	s.setupRunTask(projYml)
+	p := &model.Project{}
+	_, err := model.LoadProjectInto(s.ctx, []byte(projYml), nil, "", p)
+	s.Require().NoError(err)
+	s.tc.taskConfig.Project = p
 	s.tc.taskConfig.Task.TaskGroup = taskGroup
+	s.tc.taskConfig.TaskGroup = *p.FindTaskGroup(taskGroup)
 
-	s.Error(s.a.runPostTaskCommands(s.ctx, s.tc))
+	s.NoError(s.a.runPreTaskCommands(s.ctx, s.tc))
 
 	s.NoError(s.tc.logger.Close())
 	checkMockLogs(s.T(), s.mockCommunicator, s.tc.taskConfig.Task.Id, []string{
-		"Running post-task commands",
+		"Running pre-task commands",
```

Comment:
What happened to the check to ensure the timeout happened?

Reply:
I"m not sure which check you are referring to but it should be reset now